client/pluginmanager/manager.go

```diff
@@ -5,6 +5,10 @@ type PluginManager interface {
 	// Run starts a plugin manager and must block until shutdown
 	Run()
 
+	// Ready returns a channel that blocks until the plugin mananger has
```

Comment:
I think a simpler interface would be for Run to return when it has initialized all the plugins. This doesn't mean it has fingerprinted, but it means that all calls to plugins will be valid at that point (can dispense and use the plugin).

We can then remove `Ready()` and replace it with, `WaitForFirstFingerprint(ctx)`. So the workflow can become, the client starts and run RegisterAndRun(). Once this returns the plugin from that manager are fully usable. 

I would then add a wrapper to our current update handlers (updateNodeFromDriver/Device/Fingerprint) to have a matching batchedUpdateNodeXXX function. What the batch version would do is collect updates from the manager and store them locally until the channel returned by WaitForFingerprint is closed. This channel would be stored on the client. Once this is closed, it will pass the most recent update to the non-batch version of the function and on all subsequent calls it will be directly pass the results. We then set the callback for the managers to the batch function.

The result of this exercise is that we allow all sources of fingerprint results a chance to initially fingerprint and we can collect all the results into a single node update that is sent to the server, rather than spamming the server with many non-batched updates.